c/tskit/trees.h

```diff
@@ -63,32 +63,42 @@ extern "C" {
 @brief The tree sequence object.
 */
 typedef struct {
+    /** @brief The number of trees in this tree sequence */
     tsk_size_t num_trees;
+    /** @brief The number of nodes that are marked as samples */
     tsk_size_t num_samples;
+    /** @brief An array of ids of nodes marked as samples */
     tsk_id_t *samples;
-    /* Does this tree sequence have time_units == "uncalibrated" */
+    /** @brief Does this tree sequence have time_units == "uncalibrated" */
     bool time_uncalibrated;
-    /* Are all genome coordinates discrete? */
+    /** @brief Are all genome coordinates discrete values? */
     bool discrete_genome;
-    /* Are all time values discrete? */
+    /** @brief Are all time values discrete values? */
     bool discrete_time;
-    /* Breakpoints along the sequence, including 0 and L. */
+    /** @brief Array of breakpoint locations along the sequence, including 0 and
+     *  L. Note that the size of this array is num_trees + 1 */
     double *breakpoints;
-    /* If a node is a sample, map to its index in the samples list */
+    /** @brief The location of each sample node in the samples list or TSK_NULL if
+     *  the node is not marked as a sample*/
     tsk_id_t *sample_index_map;
-    /* Map individuals to the list of nodes that reference them */
     tsk_id_t *individual_nodes_mem;
+    /** @brief  A ragged array of node ids corresponding to each individual */
     tsk_id_t **individual_nodes;
+    /** @brief  The lengths of the arrays in individual_nodes */
     tsk_size_t *individual_nodes_length;
-    /* For each tree, a list of sites on that tree */
     tsk_site_t *tree_sites_mem;
+    /** @brief  A ragged array of site ids for each tree */
     tsk_site_t **tree_sites;
```

Comment:
I don't think we should document these things like ``inidividual_nodes``, ``site_mutations`` etc because this ties us down to *always* keeping these arrays. We could imagine doing things in future like only mallocing memory for these if someone calls the method in question.